### PR TITLE
txscript: add/group numeric related opcode tests.

### PR DESCRIPTION
**Pull request by dnldd**
This is work towards #1331

### CODE REVIEW

File: txscript/data/script_tests.json [hunk 1]
```diff
@@ -1238,9 +1238,115 @@
...
+["", "1SUB 1", "NONE", "ERR_INVALID_STACK_OPERATION", "1SUB requires one item on the stack, stack has no items"],
+["1SUB", "-1 EQUAL", "NONE", "ERR_INVALID_STACK_OPERATION", "1SUB requires one item on the stack, stack has no items"],
+["111 1SUB", "110 EQUAL", "NONE", "OK"],
+["-1 1SUB", "-2 EQUAL", "NONE", "OK", "1SUB must decrement the provided numeric stack item by one"],
+["2147483648 1SUB", "1", "NONE", "ERR_OUT_OF_RANGE", "1SUB must fail with stack item >4 bytes"],
+["0x7a{6}", "1SUB 1", "NONE", "ERR_INVALID_STACK_OPERATION", "1SUB must fail with a non-numeric stack item"],
+
+["2MUL test coverage"],
+["2 2MUL", "4 EQUAL", "NONE", "ERR_EVAL_FALSE", "2MUL is a NOP"],
+["2", "2MUL 1", "NONE", "OK", "2MUL is NOP"],
```
Comment (davecgh) [inline txscript/data/script_tests.json]: Not cleanstack.

File: txscript/data/script_tests.json [hunk 2]
```diff
@@ -1238,9 +1238,115 @@
...
+
+["2DIV test coverage"],
+["2 2DIV", "1 EQUAL", "NONE", "ERR_EVAL_FALSE", "2DIV is a NOP"],
+["2", "2DIV 1", "NONE", "OK", "2DIV is NOP"],
```
Comment (davecgh) [inline txscript/data/script_tests.json]: Not cleanstack.

File: txscript/data/script_tests.json [hunk 3]
```diff
@@ -1238,9 +1238,112 @@
...
+["1ADD test coverage"],
+["NOP", "1ADD 1", "NONE", "ERR_INVALID_STACK_OPERATION", "1ADD requires one item on the stack and NOP must not be treated as one"],
+["", "1ADD 1", "NONE", "ERR_INVALID_STACK_OPERATION", "1ADD requires one item on the stack, stack has no items"],
+["1 1ADD", "2 EQUAL", "NONE", "OK"],
+["2147483647", "1ADD 2147483648 EQUAL", "NONE", "OK", "We can do math on 4-byte integers, and compare 5-byte ones"],
+["2147483647", "1ADD 1", "NONE", "OK"],
+["-2147483647", "1ADD 1", "NONE", "OK"],
+["2147483648 1ADD", "1", "NONE", "ERR_OUT_OF_RANGE", "1ADD must fail with stack item >4 bytes"],
+["2147483647", "1ADD 1SUB 1", "NONE", "ERR_OUT_OF_RANGE", "We cannot do math on 5-byte integers, even if the result is 4-bytes"],
+["0x7a{6}", "1ADD 1", "NONE", "ERR_INVALID_STACK_OPERATION", "1ADD must fail with a non-numeric stack item"],
```
Comment (davecgh) [inline txscript/data/script_tests.json]: This, and all of the tests like it, aren't actually testing what the comment says.  `0x7a` is not a data push operation, rather it is `OP_ROLL`, so the failure is because there are no inputs to `OP_ROLL`.

Since numerbers are just raw bytes popped from the stack and interpreted as numerics according to the script number rules, it's not actually possible to test non-numeric inputs since they are all effectively numeric.  That edge conditions, such as max 4-bytes, which the above test already handles, are the relevant tests.

Consequently, this test, and all similar, should be removed.

File: txscript/data/script_tests.json [hunk 4]
```diff
@@ -1238,9 +1238,112 @@
...
+
+["1SUB test coverage"],
+["NOP", "1SUB 1", "NONE", "ERR_INVALID_STACK_OPERATION", "1SUB requires one item on the stack and NOP must not be treated as one"],
+["", "1SUB 1", "NONE", "ERR_INVALID_STACK_OPERATION", "1SUB requires one item on the stack, stack has no items"],
+["111 1SUB", "110 EQUAL", "NONE", "OK"],
+["-1 1SUB", "-2 EQUAL", "NONE", "OK", "1SUB must decrement the provided numeric stack item by one"],
+["2147483648 1SUB", "1", "NONE", "ERR_OUT_OF_RANGE", "1SUB must fail with stack item >4 bytes"],
+["0x7a{6}", "1SUB 1", "NONE", "ERR_INVALID_STACK_OPERATION", "1SUB must fail with a non-numeric stack item"],
```
Comment (davecgh) [inline txscript/data/script_tests.json]: Another case of `OP_ROLL`.

File: txscript/data/script_tests.json [hunk 5]
```diff
@@ -1238,9 +1238,112 @@
...
+["2 2DIV", "1 EQUAL", "NONE", "ERR_EVAL_FALSE", "2DIV is a NOP"],
+["2", "2DIV DROP 1", "CLEANSTACK", "OK", "2DIV is NOP"],
+
+["NEGATE test coverage"],
+["NOP", "NEGATE 1", "NONE", "ERR_INVALID_STACK_OPERATION", "NEGATE requires an item on the stack and NOP must not be treated as one"],
+["", "NEGATE -2 EQUAL", "NONE", "ERR_INVALID_STACK_OPERATION", "NEGATE requires an item on the stack"],
+["2", "NEGATE -2 EQUAL", "NONE", "OK"],
+["-5", "NEGATE 5 EQUAL", "NONE", "OK", "NEGATE must replace the numeric stack item with its negative equivalent"],
+["2147483648", "NEGATE -2147483648 EQUAL", "NONE", "ERR_OUT_OF_RANGE", "NEGATE must fail with stack item >4 bytes"],
+["0x7a{6}", "NEGATE 1", "NONE", "ERR_INVALID_STACK_OPERATION", "NEGATE must fail with a non-numeric stack item"],
```
Comment (davecgh) [inline txscript/data/script_tests.json]: Another case of `OP_ROLL`.

File: txscript/data/script_tests.json [hunk 6]
```diff
@@ -1238,9 +1238,112 @@
...
+
+["ABS test coverage"],
+["NOP", "ABS 1", "NONE", "ERR_INVALID_STACK_OPERATION", "ABS requires an item on the stack and NOP must not be treated as one"],
+["", "ABS 0 EQUAL", "NONE", "ERR_INVALID_STACK_OPERATION", "ABS requires an item on the stack, stack has no items"],
+["ABS", "0 EQUAL", "NONE", "ERR_INVALID_STACK_OPERATION", "ABS requires an item on the stack, stack has no items"],
+["0 ABS", "0 EQUAL", "NONE", "OK"],
+["16 ABS", "16 EQUAL", "NONE", "OK"],
+["-16 ABS", "16 EQUAL", "NONE", "OK", "ABS must replace the numeric stack item with its absolute equivalent"],
+["2147483648 ABS", "2147483648 EQUAL", "NONE", "ERR_OUT_OF_RANGE", "ABS must fail with stack item >4 bytes"],
+["0x7a{6}", "ABS 1", "NONE", "ERR_INVALID_STACK_OPERATION", "ABS must fail with a non-numeric stack item"],
```
Comment (davecgh) [inline txscript/data/script_tests.json]: Another case of `OP_ROLL`.

File: txscript/data/script_tests.json [hunk 7]
```diff
@@ -1238,9 +1238,112 @@
...
+
+["0NOTEQUAL test coverage"],
+["", "0NOTEQUAL 0 EQUAL", "NONE", "ERR_INVALID_STACK_OPERATION", "0NOTEQUAL requires an item on the stack, stack has no items"],
+["NOP", "0NOTEQUAL 1", "NONE", "ERR_INVALID_STACK_OPERATION", "0NOTEQUAL requires an item on the stack and NOP must not be treated as one"],
+["0 0NOTEQUAL", "0 EQUAL", "NONE", "OK"],
+["1 0NOTEQUAL", "1 EQUAL", "NONE", "OK", "0NOTEQUAL must assert if a numeric stack item is not equal to zero."],
+["111 0NOTEQUAL", "1 EQUAL", "NONE", "OK"],
+["-111 0NOTEQUAL", "1 EQUAL", "NONE", "OK"],
+["2147483648", "0NOTEQUAL 0 EQUAL", "NONE", "ERR_OUT_OF_RANGE", "0NOTEQUAL must fail with stack item >4 bytes"],
+["0x7a{6} 0NOTEQUAL", "0 EQUAL", "NONE", "ERR_INVALID_STACK_OPERATION", "0NOTEQUAL must fail with a non-numeric stack item"],
```
Comment (davecgh) [inline txscript/data/script_tests.json]: Another case of `OP_ROLL`.

File: txscript/data/script_tests.json [hunk 8]
```diff
@@ -1238,9 +1238,112 @@
...
+["2 -2 ADD", "0 EQUAL", "NONE", "OK", "ADD must add the first and second stack items"],
+["2147483647 -2147483647 ADD", "0 EQUAL", "NONE", "OK"],
+["2147483647 DUP ADD", "4294967294 EQUAL", "NONE", "OK", ">32 bit EQUAL is valid"],
+["2147483647 NEGATE DUP ADD", "-4294967294 EQUAL", "NONE", "OK"],
+["-1 -1 ADD", "-2 EQUAL", "NONE", "OK"],
+["1", "ADD", "NONE", "ERR_INVALID_STACK_OPERATION"],
+["2147483648 0 ADD", "NOP", "NONE", "ERR_OUT_OF_RANGE", "Arithmetic operands must be in range [-2^31...2^31] "],
+["-2147483648 0 ADD", "NOP", "NONE", "ERR_OUT_OF_RANGE", "Arithmetic operands must be in range [-2^31...2^31] "],
+["2147483648 1", "ADD 2147483649 EQUAL", "NONE", "ERR_OUT_OF_RANGE", "ADD must fail with stack item >4 bytes"],
+["0x7a{6} 1", "ADD 0 EQUAL", "NONE", "ERR_INVALID_STACK_OPERATION", "ADD must fail with a non-numeric stack item"],
```
Comment (davecgh) [inline txscript/data/script_tests.json]: Another case of `OP_ROLL`.

File: txscript/data/script_tests.json [hunk 9]
```diff
@@ -1238,9 +1238,112 @@
...
+["NOP", "SUB 0 EQUAL", "NONE", "ERR_INVALID_STACK_OPERATION", "SUB requires two items on the stack and NOP must not be treated as one"],
+["", "SUB 0 EQUAL", "NONE", "ERR_INVALID_STACK_OPERATION", "SUB requires two items on the stack, stack has no items"],
+["0", "SUB 0 EQUAL", "NONE", "ERR_INVALID_STACK_OPERATION", "SUB requires two items on the stack, stack has one item"],
+["2 -2 SUB", "4 EQUAL", "NONE", "OK", "SUB must subtract the first stack item from the second stack item"],
+["2147483647 2147483647 SUB", "0 EQUAL", "NONE", "OK"],
+["2147483647 2147483647 SUB", "0 EQUAL", "NONE", "OK"],
+["111 1 ADD 12 SUB", "100 EQUAL", "NONE", "OK"],
+["-1 1 SUB", "-2 EQUAL", "NONE", "OK"],
+["2147483648 1", "SUB 2147483647 EQUAL", "NONE", "ERR_OUT_OF_RANGE", "SUB must fail with stack item >4 bytes"],
+["0x7a{6} 1", "SUB 0 EQUAL", "NONE", "ERR_INVALID_STACK_OPERATION", "SUB must fail with a non-numeric stack item"],
```
Comment (davecgh) [inline txscript/data/script_tests.json]: Another case of `OP_ROLL`.

File: txscript/data/script_tests.json [hunk 10]
```diff
@@ -1238,9 +1238,112 @@
...
+["MUL test coverage"],
+["NOP", "MUL 0 EQUAL", "NONE", "ERR_INVALID_STACK_OPERATION", "MUL requires two items on the stack and NOP must not be treated as one"],
+["", "MUL 0 EQUAL", "NONE", "ERR_INVALID_STACK_OPERATION", "MUL requires two items on the stack, stack has no items"],
+["2", "MUL 0 EQUAL", "NONE", "ERR_INVALID_STACK_OPERATION", "MUL requires two items on the stack, stack has one item"],
+["2 DUP MUL", "4 EQUAL", "NONE", "OK", "MUL must multiply the first stack item with the second stack item"],
+["2 0 MUL", "0 EQUAL", "NONE", "OK"],
+["254 1 MUL", "254 EQUAL", "NONE", "OK"],
+["-12 12 MUL", "-144 EQUAL", "NONE", "OK"],
+["2147483648 2", "MUL", "NONE", "ERR_OUT_OF_RANGE", "MUL must fail with stack item >4 bytes"],
+["0x7a{6} 1", "MUL 0 EQUAL", "NONE", "ERR_INVALID_STACK_OPERATION", "MUL must fail with a non-numeric stack item"],
```
Comment (davecgh) [inline txscript/data/script_tests.json]: Another case of `OP_ROLL`.

File: txscript/data/script_tests.json [hunk 11]
```diff
@@ -1365,87 +1459,193 @@
...
+["0 BOOLOR", "NOT", "NONE", "ERR_INVALID_STACK_OPERATION", "BOOLOR requires two items on the stack, stack has no items"],
+["1 1 BOOLOR", "NOP", "NONE", "OK", "BOOLOR must perform a logical and operation the boolean values of the first two stack items"],
 ["1 0 BOOLOR", "NOP", "NONE", "OK"],
 ["0 1 BOOLOR", "NOP", "NONE", "OK"],
 ["0 0 BOOLOR", "NOT", "NONE", "OK"],
-["16 17 BOOLOR", "NOP", "NONE", "OK"],
-["11 10 1 ADD", "NUMEQUAL", "NONE", "OK"],
-["11 10 1 ADD", "NUMEQUALVERIFY 1", "NONE", "OK"],
-["11 10 1 ADD", "NUMNOTEQUAL NOT", "NONE", "OK"],
+["1 2147483648 BOOLOR", "NOT", "NONE", "ERR_OUT_OF_RANGE", "BOOLAND must fail with stack item >4 bytes"],
```
Comment (davecgh) [inline txscript/data/script_tests.json]: Same here regarding needing another test for the other operand being >4 bytes and comment.

File: txscript/data/script_tests.json [hunk 12]
```diff
@@ -1365,87 +1459,193 @@
...
+["-200 200", "NUMEQUAL NOT", "NONE", "OK"],
+["-1 -1", "NUMEQUAL", "NONE", "OK"],
+["1", "0x02 0x0100 NUMEQUAL", "NONE", "ERR_MINIMAL_DATA", "Numerically equal but rejected by minimal data encoding requirements"],
+["11", "0x4c 0x03 0x0b0000 NUMEQUAL", "NONE", "ERR_MINIMAL_DATA"],
+["0", "0x01 0x80 NUMEQUAL", "NONE", "ERR_MINIMAL_DATA", "Zero numerically equals negative zero but rejected by minimal data encoding requirements"],
+["0", "0x02 0x0080 NUMEQUAL", "NONE", "ERR_MINIMAL_DATA", "Zero numerically equals negative zero with padding but rejected by minimal data encoding requirements"],
+["0x03 0x000080", "0x04 0x00000080 NUMEQUAL", "NONE", "ERR_MINIMAL_DATA"],
+["0x03 0x100080", "0x04 0x10000080 NUMEQUAL", "NONE", "ERR_MINIMAL_DATA"],
+["0x03 0x100000", "0x04 0x10000000 NUMEQUAL", "NONE", "ERR_MINIMAL_DATA"],
+["1 2147483648 NUMEQUAL", "NOT", "NONE", "ERR_OUT_OF_RANGE", "NUMEQUAL must fail with stack item >4 bytes"],
```
Comment (davecgh) [inline txscript/data/script_tests.json]: Same here regarding needing another test for the other operand being >4 bytes and comment.

File: txscript/data/script_tests.json [hunk 13]
```diff
@@ -1365,87 +1459,193 @@
...
+["1 2147483648 BOOLOR", "NOT", "NONE", "ERR_OUT_OF_RANGE", "BOOLAND must fail with stack item >4 bytes"],
+
+["NUMEQUAL test coverage"],
+["NOP", "NUMEQUAL", "NONE", "ERR_INVALID_STACK_OPERATION", "NUMEQUAL requires two items on the stack and NOP must not be treated as one"],
+["", "NUMEQUAL", "NONE", "ERR_INVALID_STACK_OPERATION", "NUMEQUAL requires two items on the stack, stack has no items"],
+["-1", "NUMEQUAL", "NONE", "ERR_INVALID_STACK_OPERATION", "NUMEQUAL requires two items on the stack, stack has one item"],
+["11 10 1 ADD", "NUMEQUAL", "NONE", "OK", "NUMEQUAL must compare the first numeric stack item to the second numeric stack item and check if they are equal"],
+["-200 200", "NUMEQUAL NOT", "NONE", "OK"],
+["-1 -1", "NUMEQUAL", "NONE", "OK"],
+["1", "0x02 0x0100 NUMEQUAL", "NONE", "ERR_MINIMAL_DATA", "Numerically equal but rejected by minimal data encoding requirements"],
```
Comment (davecgh) [inline txscript/data/script_tests.json]: These tests for numeric equivalency with rejection due to minimal data encoding should be moved down to the section just below that is specifically dedicated to that.

File: txscript/data/script_tests.json [hunk 14]
```diff
@@ -1365,87 +1459,193 @@
...
+["0x04 0xffff7f80", "0x03 0xffffff NUMEQUAL", "NONE", "ERR_MINIMAL_DATA", "0xffff7f80 numequals 0xffffff but rejected by minimal data encoding requirements"],
+["0x04 0xffff7f00", "0x03 0xffff7f NUMEQUAL", "NONE", "ERR_MINIMAL_DATA", "0xffff7f00 numequals 0xffff7f but rejected by minimal data encoding requirements"],
+
+["NUMEQUALVERIFY test coverage"],
+["NOP NUMEQUALVERIFY", "1", "NONE", "ERR_INVALID_STACK_OPERATION", "NUMEQUALVERIFY requires two stack items and NOP must not be treated as one"],
+["NUMEQUALVERIFY", "1", "NONE", "ERR_INVALID_STACK_OPERATION", "NUMEQUALVERIFY requires two stack items, stack has no items"],
+["0 NUMEQUALVERIFY", "1", "NONE", "ERR_INVALID_STACK_OPERATION", "NUMEQUALVERIFY requires two stack items, stack has one item"],
+["11 10 1 ADD", "NUMEQUALVERIFY 1", "NONE", "OK", "NUMEQUALVERIFY must compare the first numeric stack item to the second numeric stack item, check if they are equal and verify the result"],
+["-1 -1", "NUMEQUALVERIFY 1", "NONE", "OK"],
+["1 2147483648", "NUMEQUALVERIFY 1", "NONE", "ERR_OUT_OF_RANGE", "NUMEQUALVERIFY must fail with stack item >4 bytes"],
```
Comment (davecgh) [inline txscript/data/script_tests.json]: Same here regarding needing another test for the other operand being >4 bytes and comment.

File: txscript/data/script_tests.json [hunk 15]
```diff
@@ -1365,87 +1459,193 @@
...
+
+["NUMNOTEQUAL test coverage"],
+["NOP", "NUMNOTEQUAL", "NONE", "ERR_INVALID_STACK_OPERATION", "NUMNOTEQUAL requires two stack items and NOP must not be treated as one"],
+["", "NUMNOTEQUAL", "NONE", "ERR_INVALID_STACK_OPERATION", "NUMNOTEQUAL requires two stack items, stack has no items"],
+["0", "NUMNOTEQUAL", "NONE", "ERR_INVALID_STACK_OPERATION", "NUMNOTEQUAL requires two stack items, stack has one item"],
+["11 10 1 ADD", "NUMNOTEQUAL NOT", "NONE", "OK", "NUMNOTEQUAL must compare the first numeric stack item to the second numeric stack item and check if they are not equal" ],
 ["111 10 1 ADD", "NUMNOTEQUAL", "NONE", "OK"],
-["11 10", "LESSTHAN NOT", "NONE", "OK"],
+["1 -1", "NUMNOTEQUAL", "NONE", "OK"],
+["1 2147483648", "NUMNOTEQUAL", "NONE", "ERR_OUT_OF_RANGE", "NUMNOTEQUAL must fail with stack item >4 bytes"],
```
Comment (davecgh) [inline txscript/data/script_tests.json]: Same here regarding needing another test for the other operand being >4 bytes and comment.

File: txscript/data/script_tests.json [hunk 16]
```diff
@@ -1365,87 +1459,193 @@
...
+["NOP", "LESSTHAN", "NONE", "ERR_INVALID_STACK_OPERATION", "LESSTHAN requires two items on the stack and NOP must not be treated as one"],
+["", "LESSTHAN", "NONE", "ERR_INVALID_STACK_OPERATION", "LESSTHAN requires two items on the stack, stack has no items"],
+["0", "LESSTHAN", "NONE", "ERR_INVALID_STACK_OPERATION", "LESSTHAN requires two items on the stack, stack has one item"],
+["11 10", "LESSTHAN NOT", "NONE", "OK", "LESSTHAN must check if the first numeric stack item is less than the second numeric stack item"],
 ["4 4", "LESSTHAN NOT", "NONE", "OK"],
 ["10 11", "LESSTHAN", "NONE", "OK"],
 ["-11 11", "LESSTHAN", "NONE", "OK"],
 ["-11 -10", "LESSTHAN", "NONE", "OK"],
-["11 10", "GREATERTHAN", "NONE", "OK"],
+["1 2147483648", "LESSTHAN", "NONE", "ERR_OUT_OF_RANGE", "LESSTHAN must fail with stack item >4 bytes"],
```
Comment (davecgh) [inline txscript/data/script_tests.json]: Same here regarding needing another test for the other operand being >4 bytes and comment.

File: txscript/data/script_tests.json [hunk 17]
```diff
@@ -1365,87 +1459,193 @@
...
+["NOP", "GREATERTHAN", "NONE", "ERR_INVALID_STACK_OPERATION", "GREATERTHAN requires two items on the stack and NOP must not be treated as one"],
+["", "GREATERTHAN", "NONE", "ERR_INVALID_STACK_OPERATION", "GREATERTHAN requires two items on the stack, stack has no items"],
+["0", "GREATERTHAN", "NONE", "ERR_INVALID_STACK_OPERATION", "GREATERTHAN requires two items on the stack, stack has one item"],
+["11 10", "GREATERTHAN", "NONE", "OK", "GREATERTHAN must check if the first numeric stack item is greater than the second numeric stack item"],
 ["4 4", "GREATERTHAN NOT", "NONE", "OK"],
 ["10 11", "GREATERTHAN NOT", "NONE", "OK"],
 ["-11 11", "GREATERTHAN NOT", "NONE", "OK"],
 ["-11 -10", "GREATERTHAN NOT", "NONE", "OK"],
-["11 10", "LESSTHANOREQUAL NOT", "NONE", "OK"],
+["1 2147483648", "GREATERTHAN", "NONE", "ERR_OUT_OF_RANGE", "GREATERTHAN must fail with stack item >4 bytes"],
```
Comment (davecgh) [inline txscript/data/script_tests.json]: Same here regarding needing another test for the other operand being >4 bytes and comment.

File: txscript/data/script_tests.json [hunk 18]
```diff
@@ -1365,87 +1459,193 @@
...
+["NOP", "LESSTHANOREQUAL", "NONE", "ERR_INVALID_STACK_OPERATION", "LESSTHANOREQUAL requires two items on the stack and NOP must not be treated as one"],
+["", "LESSTHANOREQUAL", "NONE", "ERR_INVALID_STACK_OPERATION", "LESSTHANOREQUAL requires two items on the stack, stack has no items"],
+["0", "LESSTHANOREQUAL", "NONE", "ERR_INVALID_STACK_OPERATION", "LESSTHANOREQUAL requires two items on the stack, stack has one item"],
+["11 10", "LESSTHANOREQUAL NOT", "NONE", "OK", "LESSTHANOREQUAL must check if the first numeric stack item is less than or equal to the second numeric stack item"],
 ["4 4", "LESSTHANOREQUAL", "NONE", "OK"],
 ["10 11", "LESSTHANOREQUAL", "NONE", "OK"],
 ["-11 11", "LESSTHANOREQUAL", "NONE", "OK"],
 ["-11 -10", "LESSTHANOREQUAL", "NONE", "OK"],
-["11 10", "GREATERTHANOREQUAL", "NONE", "OK"],
+["1 2147483648", "LESSTHANOREQUAL", "NONE", "ERR_OUT_OF_RANGE", "LESSTHANOREQUAL must fail with stack item >4 bytes"],
```
Comment (davecgh) [inline txscript/data/script_tests.json]: Same here regarding needing another test for the other operand being >4 bytes and comment.

File: txscript/data/script_tests.json [hunk 19]
```diff
@@ -1365,87 +1459,193 @@
...
+["NOP", "GREATERTHANOREQUAL", "NONE", "ERR_INVALID_STACK_OPERATION", "GREATERTHANOREQUAL requires two items on the stack and NOP must not be treated as one"],
+["", "GREATERTHANOREQUAL", "NONE", "ERR_INVALID_STACK_OPERATION", "GREATERTHANOREQUAL requires two items on the stack, stack has no items"],
+["0", "GREATERTHANOREQUAL", "NONE", "ERR_INVALID_STACK_OPERATION", "GREATERTHANOREQUAL requires two items on the stack, stack has one item"],
+["11 10", "GREATERTHANOREQUAL", "NONE", "OK", "GREATERTHANOREQUAL must check if the first numeric stack item is greater then or equal to the second numeric stack item"],
 ["4 4", "GREATERTHANOREQUAL", "NONE", "OK"],
 ["10 11", "GREATERTHANOREQUAL NOT", "NONE", "OK"],
 ["-11 11", "GREATERTHANOREQUAL NOT", "NONE", "OK"],
 ["-11 -10", "GREATERTHANOREQUAL NOT", "NONE", "OK"],
-["1 0 MIN", "0 NUMEQUAL", "NONE", "OK"],
+["1 2147483648", "GREATERTHANOREQUAL", "NONE", "ERR_OUT_OF_RANGE", "GREATERTHANOREQUAL must fail with stack item >4 bytes"],
```
Comment (davecgh) [inline txscript/data/script_tests.json]: Same here regarding needing another test for the other operand being >4 bytes and comment.

File: txscript/data/script_tests.json [hunk 20]
```diff
@@ -1365,87 +1459,193 @@
...
+["MIN test coverage"],
+["NOP", "MIN", "NONE", "ERR_INVALID_STACK_OPERATION", "MIN requires two items on the stack and NOP must not be treated as one"],
+["", "MIN", "NONE", "ERR_INVALID_STACK_OPERATION", "MIN requires two items on the stack, stack has no items"],
+["0", "MIN", "NONE", "ERR_INVALID_STACK_OPERATION", "MIN requires two items on the stack, stack has one item"],
+["1 0 MIN", "0 NUMEQUAL", "NONE", "OK", "MIN must compare the first two numeric stack items and return the mimimum"],
 ["0 1 MIN", "0 NUMEQUAL", "NONE", "OK"],
 ["-1 0 MIN", "-1 NUMEQUAL", "NONE", "OK"],
 ["0 -2147483647 MIN", "-2147483647 NUMEQUAL", "NONE", "OK"],
-["2147483647 0 MAX", "2147483647 NUMEQUAL", "NONE", "OK"],
+["1 2147483648", "MIN", "NONE", "ERR_OUT_OF_RANGE", "MIN must fail with stack >4 bytes"],
```
Comment (davecgh) [inline txscript/data/script_tests.json]: Same here regarding needing another test for the other operand being >4 bytes and comment.

File: txscript/data/script_tests.json [hunk 21]
```diff
@@ -1365,87 +1459,193 @@
...
+["MAX test coverage"],
+["NOP", "MAX", "NONE", "ERR_INVALID_STACK_OPERATION", "MAX requires two items on the stack and NOP must not be treated as one"],
+["", "MAX", "NONE", "ERR_INVALID_STACK_OPERATION", "MAX requires two items on the stack, stack has no items"],
+["0", "MAX", "NONE", "ERR_INVALID_STACK_OPERATION", "MAX requires two items on the stack, stack has one item"],
+["2147483647 0 MAX", "2147483647 NUMEQUAL", "NONE", "OK", "MAX must compare the first two numeric stack items and return the maximum"],
 ["0 100 MAX", "100 NUMEQUAL", "NONE", "OK"],
 ["-100 0 MAX", "0 NUMEQUAL", "NONE", "OK"],
 ["0 -2147483647 MAX", "0 NUMEQUAL", "NONE", "OK"],
-["0 0 1", "WITHIN", "NONE", "OK"],
+["1 2147483648", "MAX", "NONE", "ERR_OUT_OF_RANGE", "MAX must fail with stack item >4 bytes"],
```
Comment (davecgh) [inline txscript/data/script_tests.json]: Same here regarding needing another test for the other operand being >4 bytes and comment.

File: txscript/data/script_tests.json [hunk 22]
```diff
@@ -1365,87 +1459,193 @@
...
+["0", "WITHIN", "NONE", "ERR_INVALID_STACK_OPERATION", "WITHIN requires three items on the stack, stack has one item"],
+["0 1", "WITHIN", "NONE", "ERR_INVALID_STACK_OPERATION", "WITHIN requires three items on the stack, stack has two items"],
+["0 0 1", "WITHIN", "NONE", "OK", "WITHIN must check if the third numeric stack item falls within the range of the second and first numeric stack items (left inclusive)"],
 ["1 0 1", "WITHIN NOT", "NONE", "OK"],
 ["0 -2147483647 2147483647", "WITHIN", "NONE", "OK"],
 ["-1 -100 100", "WITHIN", "NONE", "OK"],
 ["11 -100 100", "WITHIN", "NONE", "OK"],
 ["-2147483647 -100 100", "WITHIN NOT", "NONE", "OK"],
 ["2147483647 -100 100", "WITHIN NOT", "NONE", "OK"],
+["1 0 2147483648", "WITHIN", "NONE", "ERR_OUT_OF_RANGE", "WITHIN must fail with stack item >4 bytes"],
```
Comment (davecgh) [inline txscript/data/script_tests.json]: Same here regarding needing another test for the other operands being >4 bytes and comment.  Note that there are 3 here instead of 2 as all of the others up to this point.

File: txscript/data/script_tests.json [hunk 23]
```diff
@@ -1238,9 +1238,104 @@
...
+["0", "ADD 0 EQUAL", "NONE", "ERR_INVALID_STACK_OPERATION", "ADD requires two items on the stack, stack has one item"],
+["2 -2 ADD", "0 EQUAL", "NONE", "OK", "ADD must add the first and second stack items"],
+["2147483647 -2147483647 ADD", "0 EQUAL", "NONE", "OK"],
+["2147483647 DUP ADD", "4294967294 EQUAL", "NONE", "OK", ">32 bit EQUAL is valid"],
+["2147483647 NEGATE DUP ADD", "-4294967294 EQUAL", "NONE", "OK"],
+["-1 -1 ADD", "-2 EQUAL", "NONE", "OK"],
+["1", "ADD", "NONE", "ERR_INVALID_STACK_OPERATION"],
+["2147483648 0 ADD", "NOP", "NONE", "ERR_OUT_OF_RANGE", "Arithmetic operands must be in range [-2^31...2^31] (positive numeric case)"],
+["-2147483648 0 ADD", "NOP", "NONE", "ERR_OUT_OF_RANGE", "Arithmetic operands must be in range [-2^31...2^31] (negative numerica case)"],
+["2147483648 1", "ADD 2147483649 EQUAL", "NONE", "ERR_OUT_OF_RANGE", "ADD must fail with first operand >4 bytes"],
```
Comment (davecgh) [inline txscript/data/script_tests.json]: Comment is incorrect.  The first operand to the opcode is the second one in the script because they are popped in reverse order.

Same is true with all of these.

File: txscript/data/script_tests.json [hunk 24]
```diff
@@ -1256,8 +1351,9 @@
 ["6 0", "DIV TRUE", "NONE", "ERR_DIVIDE_BY_ZERO", "DIV must fail with zero divisor"],
 ["2147483648 1073741824", "DIV TRUE", "NONE", "ERR_OUT_OF_RANGE", "DIV must fail with dividend >4 bytes"],
 ["1 2147483648", "DIV TRUE", "NONE", "ERR_OUT_OF_RANGE", "DIV must fail with divisor >4 bytes"],
+["2147483648 1", "DIV TRUE", "NONE", "ERR_OUT_OF_RANGE", "DIV must fail with dividend >4 bytes"],
```
Comment (davecgh) [inline txscript/data/script_tests.json]: Already tested 2 lines above.  Not needed.

File: txscript/data/script_tests.json [hunk 25]
```diff
@@ -1272,8 +1368,9 @@
 ["6 0", "MOD TRUE", "NONE", "ERR_DIVIDE_BY_ZERO", "MOD must fail with zero divisor"],
 ["2147483648 1073741824", "MOD TRUE", "NONE", "ERR_OUT_OF_RANGE", "MOD must fail with dividend >4 bytes"],
 ["1 2147483648", "MOD TRUE", "NONE", "ERR_OUT_OF_RANGE", "MOD must fail with divisor >4 bytes"],
+["2147483648 1", "MOD TRUE", "NONE", "ERR_OUT_OF_RANGE", "MOD must fail with dividend >4 bytes"],
```
Comment (davecgh) [inline txscript/data/script_tests.json]: Already tested 2 lines above. Not needed.
